Sources/Helpers/AmenitySearcher.swift

```diff
@@ -0,0 +1,197 @@
+//
+//  AmenitySearcher.swift
+//  OsmAnd
+//
+//  Created by Max Kojin on 18/07/25.
+//  Copyright © 2025 OsmAnd. All rights reserved.
+//
+
+// OsmAnd-java/src/main/java/net/osmand/search/AmenitySearcher.java
+// git revision c3d7a8a378dc84251eb7ca72dd10958af86d2225
+
+@objcMembers
+final class AmenitySearcherRequest: NSObject {
+    var latLon: CLLocation?
+    var osmId: Int64
+    var type: String?
+    var wikidata: String?
+    var names = [String]()
+    
+    init(mapObject: OAMapObject) {
+        osmId = ObfConstants.getOsmObjectId(mapObject)
+        type = ObfConstants.getOsmEntityType(mapObject)
+        
+        if let poi = mapObject as? OAPOI {
+            latLon = poi.getLocation()
+            wikidata = poi.getWikidata()
+            if let name = poi.name {
+                names.append(name)
+            }
+            if let localizedNames = poi.localizedNames.allValues as? [String] {
+                names.append(contentsOf:localizedNames)
+            }
+        } else if let renderedObject = mapObject as? OARenderedObject {
+            latLon = renderedObject.getLocation()
+            if latLon == nil || latLon?.coordinate.latitude == 0 && latLon?.coordinate.longitude == 0 {
+                latLon = renderedObject.labelLatLon
+            }
+            if let localizedNames = renderedObject.localizedNames.allValues as? [String] {
+                names.append(contentsOf:localizedNames)
+            }
+            if let value = renderedObject.tags[WIKIDATA_TAG] as? String {
+                wikidata = value
+            }
+        } else if let stop = mapObject as? OATransportStop {
+            stop.findAmenityDataIfNeeded()
+            latLon = stop.getLocation();
+            if let name = stop.name {
+                names.append(name)
+            }
+            if let localizedNames = stop.localizedNames.allValues as? [String] {
+                names.append(contentsOf:localizedNames)
+            }
+            
+        } else {
+            latLon = mapObject.getLocation()
+            wikidata = nil
+            if let localizedNames = mapObject.localizedNames.allValues as? [String] {
+                names.append(contentsOf:localizedNames)
+            }
+        }
+    }
+    
+    convenience init(mapObject: OAMapObject, names: [String]) {
+        self.init(mapObject: mapObject)
+        self.names = names
+    }
+}
+
+@objcMembers
+final class AmenitySearcher: NSObject {
+    
+    private let AMENITY_SEARCH_RADIUS = 50;
+    private let AMENITY_SEARCH_RADIUS_FOR_RELATION = 500;
+    
+    func searchAmenities(filter: OASearchPoiTypeFilter, searchLatLon: CLLocation, radius: Int, includeTravel: Bool) -> [OAPOI] {
+        OAPOIHelper.findPOI(OASearchPoiTypeFilter.acceptAll(), additionalFilter: nil, lat: searchLatLon.coordinate.latitude, lon: searchLatLon.coordinate.longitude, radius: Int32(radius), includeTravel: includeTravel, matcher: nil, publish: nil)
```

Comment:
Main goal of create AmenitySearcher is unite different searching of amenity (poi) from whole code to one place.
Here I can see OAPOIHelper.findPOI
Perhaps better to move searching to AmenitySearcher